### PR TITLE
shared-mime-info: fix post-install behaviour

### PR DESCRIPTION
**Pull request by dangowrt**
Maintainer: @flyn-org
Compile tested: aarch64/cortex-a53
Run tested: mediatek/filogic (BananaPi R4)

Description:
Move post-install script to /etc/uci-defaults so it always runs on the target and doesn't require the host to provide 'update-mime-database'.

### CODE REVIEW

File: utils/shared-mime-info/Makefile [hunk 1]
```diff
@@ -50,7 +59,8 @@ endef
 define Package/shared-mime-info/postinst
 #!/bin/sh
 
-update-mime-database /usr/share/mime/
+update-mime-database $${IPKG_INSTROOT}/usr/share/mime/
 endef
 
 $(eval $(call BuildPackage,shared-mime-info))
+$(eval $(call HostBuild))
```
Comment (BKPepe) [inline utils/shared-mime-info/Makefile:66]: I would separate adding HostPackage into the separate commit, because commit says that it fixes post install behavior.

File: utils/shared-mime-info/Makefile [hunk 2]
```diff
@@ -18,10 +18,12 @@ PKG_SOURCE:=shared-mime-info-$(PKG_VERSION).tar.bz2
 PKG_SOURCE_URL:=https://gitlab.freedesktop.org/xdg/$(PKG_NAME)/-/archive/$(PKG_VERSION)
 PKG_HASH:=32dc32ae39ff1c1bf8434dd3b36770b48538a1772bc0298509d034f057005992
 
-PKG_BUILD_DEPENDS:=gettext-full/host
+PKG_BUILD_DEPENDS:=gettext-full/host glib2/host libxml2/host shared-mime-info/host
 
 include $(INCLUDE_DIR)/package.mk
 include $(INCLUDE_DIR)/nls.mk
+include $(INCLUDE_DIR)/nls.mk
```
[the remark this answers is not in the document]
Comment (dangowrt) [reply utils/shared-mime-info/Makefile]: Oh, that was by mistake obviously. Fixed now. Thank you!

**Comment by flyn-org** [comment]
I am not sure what is going on with the automated checks. Aside from (possibly) that, this looks good.

**Comment by robimarko** [comment]
CI images were broken as they got updated to Debian 12 while everything else is using Debian 11, that was sorted out yesterday

**Comment by flyn-org** [comment]
Looks good. Recommend merge.
